### ISSUE TITLE
Compile error from field vs submodule name ambiguity when outside unopened module

### ISSUE THREAD

**Issue by TysonMN** [dotnet/fsharp#10444]
Over the years, F# has allowed more things to have the same name without getting confused.  This issue is about one such situation where F# still gets confused.

Consider the following code.

```fsharp
module Top

module A =
  type B = { C: bool }

  module B =
    module C =
      let x = 0

  let y = B.C.x

let z = A.B.C.x
//      ^^^^^
// Error FS0728 Field 'C' is not static
```

**Expected behavior**

I expected the expressions assigned to `y` and `z` to evaluation to the same thing.

**Actual behavior**

As shown in the code as a comment, the actual behavior is a compile error in the express assigned to `z` because the `B` resolves to the type `B` instead of the nested submodule `B`.

At the same time, IntelliSense gives the correct tooltip when hovering over `x`.
![2020-11-12_11-34-29_370](https://user-images.githubusercontent.com/34664007/98974969-1f824180-24db-11eb-80ee-b09d193c395e.png)

But IntelliSense also gives a tooltip that agrees with the compiler when hovering over `Field`.
![2020-11-12_11-35-19_371](https://user-images.githubusercontent.com/34664007/98975055-404a9700-24db-11eb-954e-3f5d02356e14.png)

**Known workarounds**

Open module `A` just before `z` like this.
```fsharp
open A
let z = B.C.x
```

**Related information**

* Operating system
  - Edition: Windows 10 Enterprise
  - Version: 1909
  - OS build: 18363.1198
* .NET SDK: .NET Core 3.1.403
* Visual Studio
  - Edition: Enterprise 2019
  - Version: 16.7.7

**Comment by TIHan** [comment]
Definitely feels like a bug to me, but it might be part of the design.

**Comment by TysonMN** [comment]
I ran into issue again today and struggled a bit before I remembered that I had already created this issue about it.  (This time it involved a discriminated union instead of a record.)

Any chance for progress on this?

**Comment by KevinRansom** [comment]
@dsyme , how do you want to resolve this ambuiguity?

**Comment by dsyme** [comment]
Yes, this should resolve.  As with `B.C.x` the inevitable-error of resolving to a field in `A.B.C.x` should prefer to search module B
